flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java

```diff
@@ -92,4 +92,15 @@
 					"to a JVM deadlock. See %s for details.",
 				link("https://issues.apache.org/jira/browse/FLINK-16510", "FLINK-16510"))
 				.build());
+
+	@Documentation.ExcludeFromDocumentation
+	public static final ConfigOption<Boolean> ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT = ConfigOptions
+		.key("cluster.declarative-resource-management.enabled")
+		.booleanType()
+		.defaultValue(false)
+		.withDescription("Defines whether the cluster uses declarative resource management.");
+
+	public static boolean isDeclarativeResourceManagementEnabled(Configuration configuration) {
+		return configuration.get(ENABLE_DECLARATIVE_RESOURCE_MANAGEMENT) || System.getProperties().containsKey("flink.tests.enable-declarative");
```

Comment:
What’s the reason for us introducing a system property besides the configuration property? Is this to support easier e2e testing?

Reply:
It's not for e2e testing, but allowing it to be set in the IDE.